File: algorithms/julia/matrix.jl
function matmul(A::Matrix{Int64}, B::Matrix{Int64}, n::Int)
    C = zeros(Int64, n, n)
    @inbounds for i in 1:n
        for k in 1:n
            aik = A[i, k]
            for j in 1:n
                C[i, j] += aik * B[k, j]
            end
        end
    end
    return C
end

function main()
    input = split(read(stdin, String))
    idx   = 1
    n     = parse(Int, input[idx]); idx += 1

    A = Matrix{Int64}(undef, n, n)
    B = Matrix{Int64}(undef, n, n)

    for i in 1:n, j in 1:n
        A[i,j] = parse(Int64, input[idx]); idx += 1
    end
    for i in 1:n, j in 1:n
        B[i,j] = parse(Int64, input[idx]); idx += 1
    end

    # JIT warmup
    w = ones(Int64, 2, 2)
    matmul(w, w, 2)

    start    = time_ns()
    C        = matmul(A, B, n)
    ms       = (time_ns() - start) / 1e6

    checksum = sum(C)
    println(checksum)
    println(stderr, ms)
end

main()

File: algorithms/rust/matrix.rs
use std::io::{self, Read, Write, BufWriter};
use std::time::Instant;

fn main() {
    let mut input = String::new();
    io::stdin().read_to_string(&mut input).unwrap();
    let mut iter = input.split_ascii_whitespace();

    let n: usize = iter.next().unwrap().parse().unwrap();
    let size = n * n;

    let a: Vec<i64> = (0..size)
        .map(|_| iter.next().unwrap().parse().unwrap())
        .collect();
    let b: Vec<i64> = (0..size)
        .map(|_| iter.next().unwrap().parse().unwrap())
        .collect();
    let mut c = vec![0i64; size];

    let start = Instant::now();

    for i in 0..n {
        for k in 0..n {
            let aik = a[i*n + k];
            for j in 0..n {
                c[i*n + j] += aik * b[k*n + j];
            }
        }
    }

    let ms = start.elapsed().as_secs_f64() * 1000.0;

    let checksum: i64 = c.iter().sum();
    let stdout = io::stdout();
    let mut out = BufWriter::new(stdout.lock());
    writeln!(out, "{}", checksum).unwrap();
    eprintln!("{:.3}", ms);
}